fml/message_loop_task_queues.cc

```diff
@@ -230,22 +228,30 @@ bool MessageLoopTaskQueues::Merge(TaskQueueId owner, TaskQueueId subsumed) {
   std::lock_guard guard(queue_mutex_);
   auto& owner_entry = queue_entries_.at(owner);
   auto& subsumed_entry = queue_entries_.at(subsumed);
-
-  if (owner_entry->owner_of == subsumed) {
+  auto& subsumed_set = owner_entry->owner_of;
+  if (subsumed_set.find(subsumed) != subsumed_set.end()) {
     return true;
   }
 
-  std::vector<TaskQueueId> owner_subsumed_keys = {
-      owner_entry->owner_of, owner_entry->subsumed_by, subsumed_entry->owner_of,
-      subsumed_entry->subsumed_by};
-
-  for (auto key : owner_subsumed_keys) {
-    if (key != _kUnmerged) {
-      return false;
-    }
+  // Only don't check owner_entry->owner_of, it may contains items when merged
```

Comment:
s/Only don't/Don't only/

Reply:
The comments here are misunderstood. I have revised the comments again.